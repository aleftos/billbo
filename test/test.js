/*
 * Coder & programmer: Luis Sanabria
 * Copyright: (c) Arquetic.org 2019 
 * Date: March, 2019
 * GitHub: https://github.com/aleftos/billbo
 * Summary: Utilities for the national tax billing system of Bolivian country.
 * Description: Massive CUF test
*/

const chalk = require('chalk');
const jsome = require('jsome');
const log = console.log;

const test = require('./test-data');
const cuf = require('../lib/cuf');

jsome.level.show = true;

test.testData.forEach((element, i) => {
  let data = cuf.getCUF(element);
  log(chalk.blue('Data in '), i + 1 + ': ');
  jsome(element);
  log(chalk.cyan('Reference CUF:', test.referenceCUFs[i]));
  if (data.CUF === test.referenceCUFs[i]) {
    log(chalk.green('Result: CUF OK'));
  } else {
    log(chalk.red('Result: CUF NOK'));
  }
});