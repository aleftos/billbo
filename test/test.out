Data in  1: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "1",
. FECHA_HORA: "20190113163721231",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "1",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32DBE5A2A34214B245A6A3"
}
Reference CUF: 159FFE6FB1986A24BB32DBE5A2A34214B245A6A3
Result: CUF OK
Data in  2: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "2",
. FECHA_HORA: "20190113163721238",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "2",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32EAB8592FA23EE2CA4D48"
}
Reference CUF: 159FFE6FB1986A24BB32EAB8592FA23EE2CA4D48
Result: CUF OK
Data in  3: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "3",
. FECHA_HORA: "20190113163721238",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "3",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32EAB857CFF35D787AF3E4"
}
Reference CUF: 159FFE6FB1986A24BB32EAB857CFF35D787AF3E4
Result: CUF OK
Data in  4: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "4",
. FECHA_HORA: "20190113163721239",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "4",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32ECD6730CF0AB8D2C1A89"
}
Reference CUF: 159FFE6FB1986A24BB32ECD6730CF0AB8D2C1A89
Result: CUF OK
Data in  5: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "5",
. FECHA_HORA: "20190113163721239",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "5",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32ECD671AD41CA22DCC125"
}
Reference CUF: 159FFE6FB1986A24BB32ECD671AD41CA22DCC125
Result: CUF OK
Data in  6: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "6",
. FECHA_HORA: "20190113163721239",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "6",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32ECD673141DD973A167C5"
}
Reference CUF: 159FFE6FB1986A24BB32ECD673141DD973A167C5
Result: CUF OK
Data in  7: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "7",
. FECHA_HORA: "20190113163721240",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "7",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32EEF48B8A9036CD3E8E61"
}
Reference CUF: 159FFE6FB1986A24BB32EEF48B8A9036CD3E8E61
Result: CUF OK
Data in  8: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "8",
. FECHA_HORA: "20190113163721241",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "8",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F112A6D23600D0433506"
}
Reference CUF: 159FFE6FB1986A24BB32F112A6D23600D0433506
Result: CUF OK
Data in  9: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "9",
. FECHA_HORA: "20190113163721241",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "9",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F112A572871F65F3DBA2"
}
Reference CUF: 159FFE6FB1986A24BB32F112A572871F65F3DBA2
Result: CUF OK
Data in  10: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "10",
. FECHA_HORA: "20190113163721242",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "10",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F330C0AF846D7AA50240"
}
Reference CUF: 159FFE6FB1986A24BB32F330C0AF846D7AA50240
Result: CUF OK
Data in  11: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "11",
. FECHA_HORA: "20190113163721242",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "11",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F330BF4FD58C1055A8E7"
}
Reference CUF: 159FFE6FB1986A24BB32F330BF4FD58C1055A8E7
Result: CUF OK
Data in  12: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "12",
. FECHA_HORA: "20190113163721242",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "12",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F330C0B6B19B611A4F87"
}
Reference CUF: 159FFE6FB1986A24BB32F330C0B6B19B611A4F87
Result: CUF OK
Data in  13: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "13",
. FECHA_HORA: "20190113163721243",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "13",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F54ED92D23F8BAB77628"
}
Reference CUF: 159FFE6FB1986A24BB32F54ED92D23F8BAB77628
Result: CUF OK
Data in  14: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "14",
. FECHA_HORA: "20190113163721243",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "14",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F54EDA9400080B7C1CC8"
}
Reference CUF: 159FFE6FB1986A24BB32F54EDA9400080B7C1CC8
Result: CUF OK
Data in  15: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "15",
. FECHA_HORA: "20190113163721243",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "15",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F54ED9345126A12CC364"
}
Reference CUF: 159FFE6FB1986A24BB32F54ED9345126A12CC364
Result: CUF OK
Data in  16: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "16",
. FECHA_HORA: "20190113163721244",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "16",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F76CF4714E74B5DDEA09"
}
Reference CUF: 159FFE6FB1986A24BB32F76CF4714E74B5DDEA09
Result: CUF OK
Data in  17: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "17",
. FECHA_HORA: "20190113163721244",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "17",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F76CF3119F934B8E90A5"
}
Reference CUF: 159FFE6FB1986A24BB32F76CF3119F934B8E90A5
Result: CUF OK
Data in  18: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "18",
. FECHA_HORA: "20190113163721245",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "18",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F98B0E59455D4E933741"
}
Reference CUF: 159FFE6FB1986A24BB32F98B0E59455D4E933741
Result: CUF OK
Data in  19: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "19",
. FECHA_HORA: "20190113163721245",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "19",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32F98B0CEEEDFFF5F05DE6"
}
Reference CUF: 159FFE6FB1986A24BB32F98B0CEEEDFFF5F05DE6
Result: CUF OK
Data in  20: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "20",
. FECHA_HORA: "20190113163721246",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "20",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FBA9283693C9F8F50486"
}
Reference CUF: 159FFE6FB1986A24BB32FBA9283693C9F8F50486
Result: CUF OK
Data in  21: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "21",
. FECHA_HORA: "20190113163721246",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "21",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FBA926D6E4E88EA5AB22"
}
Reference CUF: 159FFE6FB1986A24BB32FBA926D6E4E88EA5AB22
Result: CUF OK
Data in  22: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "22",
. FECHA_HORA: "20190113163721247",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "22",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FDC74213E236A356D1C7"
}
Reference CUF: 159FFE6FB1986A24BB32FDC74213E236A356D1C7
Result: CUF OK
Data in  23: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "23",
. FECHA_HORA: "20190113163721247",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "23",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FDC740B4335539077863"
}
Reference CUF: 159FFE6FB1986A24BB32FDC740B4335539077863
Result: CUF OK
Data in  24: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "24",
. FECHA_HORA: "20190113163721247",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "1",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FDC7421A3E357D7FBF04"
}
Reference CUF: 159FFE6FB1986A24BB32FDC7421A3E357D7FBF04
Result: CUF OK
Data in  25: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "25",
. FECHA_HORA: "20190113163721248",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "1",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "2",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FFE55A90B092D71CE5A5"
}
Reference CUF: 159FFE6FB1986A24BB32FFE55A90B092D71CE5A5
Result: CUF OK
Data in  26: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "26",
. FECHA_HORA: "20190113163721248",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "3",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FFE55C1B139497A28C43"
}
Reference CUF: 159FFE6FB1986A24BB32FFE55C1B139497A28C43
Result: CUF OK
Data in  27: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "26",
. FECHA_HORA: "20190113163721248",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "4",
. POS: "0",
. CUF: "159FFE6FB1986A24BB32FFE55ABB64B32D51AC44"
}
Reference CUF: 159FFE6FB1986A24BB32FFE55ABB64B32D5332E1
Result: CUF NOK
Data in  28: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "26",
. FECHA_HORA: "20190113163721249",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "5",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33020375F8620142014C43"
}
Reference CUF: 159FFE6FB1986A24BB33020375F8620142045984
Result: CUF NOK
Data in  29: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "29",
. FECHA_HORA: "20190113163721249",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "6",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3302037498B31FD7B50020"
}
Reference CUF: 159FFE6FB1986A24BB3302037498B31FD7B50020
Result: CUF OK
Data in  30: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "30",
. FECHA_HORA: "20190113163721249",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "7",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33020375FF8F2F2879A6C2"
}
Reference CUF: 159FFE6FB1986A24BB33020375FF8F2F2879A6C2
Result: CUF OK
Data in  31: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "31",
. FECHA_HORA: "20190113163721250",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "8",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3304218E76018C8216CD67"
}
Reference CUF: 159FFE6FB1986A24BB3304218E76018C8216CD67
Result: CUF OK
Data in  32: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "32",
. FECHA_HORA: "20190113163721250",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "9",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3304218FDCDD9BD2DB7407"
}
Reference CUF: 159FFE6FB1986A24BB3304218FDCDD9BD2DB7407
Result: CUF OK
Data in  33: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "33",
. FECHA_HORA: "20190113163721251",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "10",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33063FA85DF8751ACC1AA1"
}
Reference CUF: 159FFE6FB1986A24BB33063FA85DF8751ACC1AA1
Result: CUF OK
Data in  34: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "34",
. FECHA_HORA: "20190113163721251",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "11",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33063FA9BA2C087D3D4141"
}
Reference CUF: 159FFE6FB1986A24BB33063FA9BA2C087D3D4141
Result: CUF OK
Data in  35: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "35",
. FECHA_HORA: "20190113163721251",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "12",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33063FA85A7D2712EDE7E6"
}
Reference CUF: 159FFE6FB1986A24BB33063FA85A7D2712EDE7E6
Result: CUF OK
Data in  36: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "36",
. FECHA_HORA: "20190113163721252",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "13",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33085DC3A222F115F28E82"
}
Reference CUF: 159FFE6FB1986A24BB33085DC3A222F115F28E82
Result: CUF OK
Data in  37: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "37",
. FECHA_HORA: "20190113163721252",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "14",
. POS: "0",
. CUF: "159FFE6FB1986A24BB33085DC237CB93BD4FB527"
}
Reference CUF: 159FFE6FB1986A24BB33085DC237CB93BD4FB527
Result: CUF OK
Data in  38: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "38",
. FECHA_HORA: "20190113163721253",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "15",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330A7BDD7F715DC0545BC3"
}
Reference CUF: 159FFE6FB1986A24BB330A7BDD7F715DC0545BC3
Result: CUF OK
Data in  39: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "39",
. FECHA_HORA: "20190113163721253",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "16",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330A7BDC1FC27C56050261"
}
Reference CUF: 159FFE6FB1986A24BB330A7BDC1FC27C56050261
Result: CUF OK
Data in  40: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "40",
. FECHA_HORA: "20190113163721253",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "17",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330A7BDD7BF60FB8762901"
}
Reference CUF: 159FFE6FB1986A24BB330A7BDD7BF60FB8762901
Result: CUF OK
Data in  41: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "41",
. FECHA_HORA: "20190113163721254",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "18",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330C99F5FD10E90066CFA2"
}
Reference CUF: 159FFE6FB1986A24BB330C99F5FD10E90066CFA2
Result: CUF OK
Data in  42: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "42",
. FECHA_HORA: "20190113163721254",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "19",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330C99F763ECF8512B7642"
}
Reference CUF: 159FFE6FB1986A24BB330C99F763ECF8512B7642
Result: CUF OK
Data in  43: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "43",
. FECHA_HORA: "20190113163721254",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "20",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330C99F5F9959AF8889CE9"
}
Reference CUF: 159FFE6FB1986A24BB330C99F5F9959AF8889CE9
Result: CUF OK
Data in  44: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "44",
. FECHA_HORA: "20190113163721255",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "21",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330EB811413B64FB8D4385"
}
Reference CUF: 159FFE6FB1986A24BB330EB811413B64FB8D4385
Result: CUF OK
Data in  45: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "45",
. FECHA_HORA: "20190113163721255",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "22",
. POS: "0",
. CUF: "159FFE6FB1986A24BB330EB80FE18C83913DEA21"
}
Reference CUF: 159FFE6FB1986A24BB330EB80FE18C83913DEA21
Result: CUF OK
Data in  46: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "46",
. FECHA_HORA: "20190113163721256",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "23",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3310D62B1E89D1A5EF10C6"
}
Reference CUF: 159FFE6FB1986A24BB3310D62B1E89D1A5EF10C6
Result: CUF OK
Data in  47: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "47",
. FECHA_HORA: "20190113163721256",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "1",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3310D629BE09C12F535763"
}
Reference CUF: 159FFE6FB1986A24BB3310D629BE09C12F535763
Result: CUF OK
Data in  48: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "48",
. FECHA_HORA: "20190113163721257",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "3",
. TIPO_DOCUMENTO_SECTOR: "2",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3312F44505AF8B3257FE01"
}
Reference CUF: 159FFE6FB1986A24BB3312F44505AF8B3257FE01
Result: CUF OK
Data in  49: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "49",
. FECHA_HORA: "20190113163721257",
. SUCURSAL: "0",
. MODALIDAD: "1",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "1",
. TIPO_DOCUMENTO_SECTOR: "3",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3312F4439B582DD9B524A4"
}
Reference CUF: 159FFE6FB1986A24BB3312F4439B582DD9B524A4
Result: CUF OK
Data in  50: 
{
. NIT_EMISOR: "123456789",
. NÚMERO_FACTURA: "50",
. FECHA_HORA: "20190113163721258",
. SUCURSAL: "0",
. MODALIDAD: "2",
. TIPO_EMISIÓN: "2",
. CÓDIGO_DOCUMENTO_FISCAL: "2",
. TIPO_DOCUMENTO_SECTOR: "4",
. POS: "0",
. CUF: "159FFE6FB1986A24BB3315125EE2FDF7DCB9CB42"
}
Reference CUF: 159FFE6FB1986A24BB3315125EE2FDF7DCB9CB42
Result: CUF OK
